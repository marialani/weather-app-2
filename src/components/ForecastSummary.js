import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import WeatherIcon from "react-icons-weather";
import "../styles/ForecastSummary.css";

const ForecastSummary = (props) => {
  const { date, icon, temperature, description, onSelect } = props;
  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <WeatherIcon name="owm" iconId={icon} />
      </div>
      <div className="forecast-summary__temperature">
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-summary__description">{description}</div>
      <div className="forecast-details__button">
        <button
          className="moreDetails__button"
          type="button"
          onClick={() => onSelect(date)}
        >
          More details
        </button>
      </div>
    </div>
  );
};

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
  onSelect: PropTypes.func.isRequired,
};

export default ForecastSummary;
