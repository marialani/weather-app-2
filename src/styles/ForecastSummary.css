.moreDetails__button {
    display: flex;
    flex-direction: row;
    padding: 12px 29px;
    margin: 0.5rem;
    background-color: white;
    color: black;
    border: 2px solid #002E45;
    border-radius: 10px;
    text-align: center;
    font-size: 15px;
    box-shadow: rgba(0, 0, 0, 0.4) 0px 2px 4px, rgba(0, 0, 0, 0.3) 0px 7px 13px -3px, rgba(0, 0, 0, 0.2) 0px -3px 0px inset;

}

.moreDetails__button:hover {
    border: 2px solid #2B7994;
    color: #2B7994;
    cursor: pointer;
    
}

.forecast-summary {
    font-weight: bold;
    padding: 0.5rem;
    text-align: center;
}

 .forecast-summary__icon {
    padding: 0.5rem;
   
} 

.forecast-summary__date {
    font-size: 25px;
}



