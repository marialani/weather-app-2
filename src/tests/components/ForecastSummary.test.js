import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 3333444,
    description: "Stub description",
    icon: "800",
    temperature: {
      min: 20,
      max: 30,
    },
    onSelect: () => {},
  };
  it("renders details of the forecast including: date, description, icon and temperature", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        onSelect={validProps.onSelect}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        onSelect={validProps.onSelect}
      />
    );
    expect(getByText("Thu 1st Jan")).toHaveClass("forecast-summary__date");
    expect(getByText("Stub description")).toHaveClass(
      "forecast-summary__description"
    );
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
    expect(getByText("30Â°C")).toHaveClass("forecast-summary__temperature");
  });
});
