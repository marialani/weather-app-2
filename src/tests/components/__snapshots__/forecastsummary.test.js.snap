// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders details of the forecast including: date, description, icon and temperature 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      Thu 1st Jan
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-sunny"
        name="owm"
      />
    </div>
    <div
      class="forecast-summary__temperature"
    >
      30Â°C
    </div>
    <div
      class="forecast-summary__description"
    >
      Stub description
    </div>
    <div
      class="forecast-details__button"
    >
      <button
        class="moreDetails__button"
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
